agent/connect/ca/provider_vault.go

```diff
@@ -21,12 +24,18 @@ var ErrBackendNotMounted = fmt.Errorf("backend not mounted")
 var ErrBackendNotInitialized = fmt.Errorf("backend not initialized")
 
 type VaultProvider struct {
-	config                       *structs.VaultCAProviderConfig
-	client                       *vaultapi.Client
+	config *structs.VaultCAProviderConfig
+	client *vaultapi.Client
+
+	shutdown     bool
+	shutdownCh   chan struct{}
+	shutdownLock sync.RWMutex
```

Comment:
I noticed this pattern in a couple other places in Consul, and I've been wondering if there is a way to do this without all 3 fields. I think we've also been slowing moving things to use `context.Context` to manage the lifecycle of goroutines.  I suspect in this case something like this might work:

A field of `shutdown func()` that is initialized to `func() {}` in `Server.createCAProvider()` where it is created, so that it can never be nil, or maybe in a new `newVaultProvider` func if we need to initialize it in tests as well.

In `Configure` the context can be passed to `renewToken`

```go

ctx, cancel := context.WithCancel(context.TODO())
v.shutdown = cancel
go renewToken(ctx, renewer)
```

`renewToken` can use `<- ctx.Done()` instead of the `shutdownCh`, and `Stop` only needs to call `v.shutdown()`, because the cancel function is safe to call multiple times, and is never nil.